### ISSUE TITLE
error while creating mount source path '/var/lib/etcd': mkdir /var/lib/etcd: permission denied

### ISSUE THREAD

**Issue by StefanWallin** [rancher/rancher#13900]
**Rancher versions:**
rancher/rancher: latest (2018-06-08: v2.0.2)
rancher/rancher-agent: v2.0.2

**Infrastructure Stack versions:**
healthcheck: ?
ipsec: ?
network-services: ?
scheduler: ?
kubernetes (if applicable): ?

**Docker version: (`docker version`,`docker info` preferred)**

*Machine 1: Debian Jessie, Docker 18.03.1-ce, hostname: `banana`, ip: `192.168.8.227`, purpose: rancher ui*
```
Containers: 2
 Running: 1
 Paused: 0
 Stopped: 1
Images: 1
Server Version: 18.03.1-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 14
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 773c489c9c1b21a6d78b5c538cd395416ec50f88
runc version: 4fc53a81fb7c994640722ac585fa9ca548971871
init version: 949e6fa
Kernel Version: 3.16.0-4-amd64
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: x86_64
CPUs: 6
Total Memory: 3.777GiB
Name: banana
ID: OJVT:RUUO:7A3D:GCF7:3U57:VFO6:LZGN:4KJK:VA6X:O4SD:MULO:HC6L
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No memory limit support
WARNING: No swap limit support
WARNING: No kernel memory limit support
WARNING: No oom kill disable support
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
```

*Machine 2: Ubuntu bionic, Docker 17.06.2-c2, hostname: `honeydew`, ip: `192.168.8.220` purpose: rancher agent*
```
Containers: 5
 Running: 2
 Paused: 0
 Stopped: 3
Images: 4
Server Version: 17.06.2-ce
Storage Driver: aufs
 Root Dir: /var/snap/docker/common/var-lib-docker/aufs
 Backing Filesystem: extfs
 Dirs: 47
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 6e23458c129b551d5c9871e5174f6b1b7f6d1170
runc version: 810190ceaa507aa2727d7ae6f4790c76ec150bd2
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.15.0-22-generic
Operating System: Ubuntu Core 16
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.544GiB
Name: honeydew
ID: BADJ:GLYH:QN2J:QESQ:VEEP:ULGW:APNG:2D3P:MRVQ:EGO2:TZHM:HVV6
Docker Root Dir: /var/snap/docker/common/var-lib-docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 42
 Goroutines: 49
 System Time: 2018-06-08T07:30:05.458371774Z
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```
**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
*Machine 1: Debian Jessie, Docker 18.03.1-ce, hostname: `banana`, ip: `192.168.8.227` purpose: rancher ui*
```
PRETTY_NAME="Debian GNU/Linux 8 (jessie)"
NAME="Debian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=debian
HOME_URL="http://www.debian.org/"
SUPPORT_URL="http://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```
*Machine 2: Ubuntu bionic, Docker 17.06.2-c2, hostname: `honeydew`, ip: `192.168.8.220` purpose: rancher agent*
```
NAME="Ubuntu"
VERSION="18.04 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```
**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single node rancher/rancher + Single node rancher/agent

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
?

**Steps to Reproduce:**
*Machine 1: Debian Jessie, Docker 18.03.1-ce, hostname: `banana`, ip: `192.168.8.227` purpose: rancher ui*
- run `sudo docker run -d --restart=unless-stopped -p 80:80 -p 443:443 rancher/rancher`
- Add custom cluster, only give name according to [Quick Start Guide](https://rancher.com/docs/rancher/v2.x/en/quick-start-guide/)
- Select all three node roles, copy comand

*Machine 2: Ubuntu bionic, Docker 17.06.2-c2, hostname: `honeydew`, ip: `192.168.8.220` purpose: rancher agent*
- run `sudo docker run -d --privileged --restart=unless-stopped --net=host -v /etc/kubernetes:/etc/kubernetes -v /var/run:/var/run rancher/rancher-agent:v2.0.2 --server https://banana --token ...omitted... --ca-checksum d177d1abd136f16ea83e89bede2dcc1b24416854c055e1ce7d2685eefa5fb8f5 --etcd --controlplane --worker`

**Results:**
- While provisioning the first node on Machine 2(rancher agent), the rancher ui shows this:
<img width="1268" alt="screen shot 2018-06-08 at 09 22 59" src="https://user-images.githubusercontent.com/457653/41145057-274e1e7c-6aff-11e8-9723-6a8345f66c99.png">
- The error message in rancher ui in plain text

```
This cluster is currently Provisioning; areas that interact directly with it will not be available until the API is ready.

[etcd] Failed to bring up Etcd Plane: Failed to start [etcd] container on host [192.168.8.220]: Error response from daemon: error while creating mount source path '/var/lib/etcd': mkdir /var/lib/etcd: permission denied
```

- The logs in honeydew for the first docker image show:
```
...
-----END CERTIFICATE-----
time="2018-06-08T07:17:38Z" level=info msg="Option etcd=true"
time="2018-06-08T07:17:38Z" level=info msg="Option controlPlane=true"
time="2018-06-08T07:17:38Z" level=info msg="Option worker=true"
time="2018-06-08T07:17:38Z" level=info msg="Option requestedHostname=honeydew"
time="2018-06-08T07:17:38Z" level=info msg="Option customConfig=map[roles:[etcd worker controlplane] address:192.168.8.220 internalAddress:]"
time="2018-06-08T07:17:38Z" level=info msg="Connecting to wss://banana/v3/connect/register with token ...omitted..."
time="2018-06-08T07:17:38Z" level=info msg="Connecting to proxy" url="wss://banana/v3/connect/register"
time="2018-06-08T07:17:38Z" level=error msg="Failed to connect to proxy" error="websocket: bad handshake"
time="2018-06-08T07:17:38Z" level=error msg="Failed to connect to proxy" error="websocket: bad handshake"
time="2018-06-08T07:17:48Z" level=info msg="Connecting to wss://banana/v3/connect/register with token ...omitted..."
time="2018-06-08T07:17:48Z" level=info msg="Connecting to proxy" url="wss://banana/v3/connect/register"
time="2018-06-08T07:17:49Z" level=info msg="Starting plan monitor"
```

- The logs in honeydew for the second docker image show:
```
Found container ID: 2c61c2d7025f2e2c6866c6965303170a6a969044f40f075cbc6608f323f68e1c
Checking root: /host/run/runc
Checking file: 2867f2819704d2737e4a297410da547ce7791ee2186a9f0f5d13f315ac3ee260
Checking file: 2c61c2d7025f2e2c6866c6965303170a6a969044f40f075cbc6608f323f68e1c
Found state.json: 2c61c2d7025f2e2c6866c6965303170a6a969044f40f075cbc6608f323f68e1c
time="2018-06-08T06:47:50Z" level=info msg="Execing [/usr/bin/nsenter --mount=/proc/17508/ns/mnt -F -- /var/snap/docker/common/var-lib-docker/aufs/mnt/73146d21e8d85e7693f12549be6db0ea0cf341e59b9248feb6cb5f8b8b7f851e/usr/bin/share-mnt --stage2 /var/lib/kubelet /var/lib/rancher -- norun]"
Error response from daemon: {"message":"No such container: kubelet"}
Error: failed to start containers: kubelet
Error response from daemon: {"message":"No such container: kubelet"}
Error: failed to start containers: kubelet
Error response from daemon: {"message":"No such container: kubelet"}
Error: failed to start containers: kubelet
Error response from daemon: {"message":"No such container: kubelet"}
Error: failed to start containers: kubelet
...
```

**Comment by superseb** [comment]
There is a known issue when using `snap` to install Docker, please use upstream packages and try again.

**Comment by StefanWallin** [comment]
Thanks for the quick response! WIll try this

**Comment by StefanWallin** [comment]
Documenting here for others that have these issues:

TLDR; 
- USE SUPPORTED OS VERSIONS, SUCH AS UBUNTU 16.04, 
- [You can use debian with this fix](https://github.com/kubernetes/kubernetes-anywhere/blob/fd0670ba051bf96be335a3291dab7039a32381d7/FIXES.md#debian-8-and-cgroups)

**Longer**
I see now why, I did not find the `docker-ce` package on the `stable` package repositories for Ubuntu 18.04, only in testing, but apt-get suggested snap and I tried that which ended me up with this error.

I tried to add a `test` package repository for ubuntu for now to enable installing docker-ce from there.
That way I could get hold of `docker-ce`, but only 18.05.0, not the required 17.03.2. So it seems Ubuntu 18.04 is a no go and that Ubuntu 16.04 or some other distro is the option in this case.

Using 18.05.0 rendered other errors since kubernetes only supports up to 17.12.0 currently

**Comment by superseb** [comment]
https://github.com/rancher/rancher/issues/13888

**Comment by dhrn** [comment]
facing similar problem with zypper

**Comment by dhrn** [comment]
any idea to fix this?

**Comment by nagug** [comment]
check if the docker is installed as snap package by
`sudo snap list`

if its a snap installation, remove docker using snap

once done, Please do a fresh installation using direct docker upstream. That should ideally solve

**Comment by abserari** [comment]
`sudo snap remove docker`

use this to remove docker
